### PR TITLE
add naive model

### PR DESCRIPTION
**Pull request by LeonieFreisinger**
Implementation replaces PR #1 

**Problem**
Existing PR of naive model implementation is based on a too old version.

**Solution**
Implement naive model from scratch. 

**Key Changes**
- NaiveModel implemented as child class of SeasonalNaiveModel PR #19 

**Model capabilities**
- univariate time series, df with single ID
- n_forecasts>=1
- n_lags =0

**Generally not supported by the model**
- autoregression
- past covariates
- future covariates 

**Further open requirements**
see parent class `SeasonalNaiveModel()`

 **Review Checklist**
- [x] I have performed a self-review of my own code.
- [x]  I have commented my code, added Docstrings and data types to function definitions.
- [x] I have added model class Docstring
- [x] I have added pytests to check whether my feature / fix works and assertations are called (checked with/ without ID column)

### CODE REVIEW

File: tot/models.py [hunk 1]
```diff
@@ -225,3 +232,357 @@ def maybe_drop_added_dates(self, predicted, df):
...
+        return fcst_df
+
+
+@dataclass
+class NaiveModel(SeasonalNaiveModel):
+    """
+    A `NaiveModel` is a naive model that forecasts future values of a target series as the value of the
+    last observation of the target series. The NaiveModel is SeasonalNaiveModel with K=1.
+
+    The naive models act on time series data having ``fit()`` and ``predict()`` methods.
```
Comment (ourownstory) [inline tot/models.py:543]: I think you can remove most of the things below in the docstring?

Comment (ourownstory) [reply tot/models.py:543]: this should not receive any `K`, too?

File: tot/models.py [hunk 2]
```diff
@@ -225,3 +232,357 @@ def maybe_drop_added_dates(self, predicted, df):
...
+
+        # expect no _data_params
+        data_params = self.params["_data_params"]
+        assert len(data_params) == 0
+        custom_seasonalities = None  # TODO: auto-detect season from metadata
+
+        # Verify expected model_prams: K > 1, n_forecasts >1
+        model_params = deepcopy(self.params)
+        model_params.pop("_data_params")
+        assert len(model_params) == 1, "Model parameters K and n_forecasts must be provided"
```
Comment (ourownstory) [inline tot/models.py:584]: just n_forecasts, no?

File: tot/models.py [hunk 3]
```diff
@@ -225,3 +232,357 @@ def maybe_drop_added_dates(self, predicted, df):
...
+        self.n_forecasts = model_params["n_forecasts"]
+        assert self.n_forecasts > 1, "Model parameter n_forecasts must be >1. "
```
Comment (ourownstory) [inline tot/models.py:586]: why not 1?

File: tot/models.py [hunk 4]
```diff
@@ -225,3 +232,357 @@ def maybe_drop_added_dates(self, predicted, df):
...
+        self.K = 1  # set to 1 for NaiveModel
```
Comment (ourownstory) [inline tot/models.py:587]: could this be set in the pass to super?

File: tot/models.py (hunk 1 above)
Comment (ourownstory) [reply tot/models.py:543]: maybe it is simpler to reimplement it after all?

File: tests/test_models.py [hunk 5]
```diff
@@ -100,3 +101,118 @@ def test_prophet_for_global_modeling():
...
+        save_dir=SAVE_DIR,
+        num_processes=1,
+    )
+    with pytest.raises(AssertionError):
+        _, _ = benchmark.run()
+
+    log.info("#### Done with test_seasonal_naive_model")
+
+
+def test_naive_model():
```
Comment (ourownstory) [inline tests/test_models.py:163]: see comment in seasonal naive

File: tot/models.py (hunk 1 above)
Comment (LeonieFreisinger) [reply tot/models.py:543]: @ourownstory Let`s quickly discuss the docstring and reimplementation verbally. :)

**Pull request merged**